### PR TITLE
[PHI] Migrate pool2d and pool2d_grad kernels

### PR DESCRIPTION
**Pull request by piotrekobi**
### PR types

Others
### PR changes

Others
### Describe

Migrates the following oneDNN operator fluid kernels to PHI:
- pool2d
- pool2d_grad

### CODE REVIEW

**Review by YuanRisheng — DISMISSED** [review]

File: paddle/fluid/operators/mkldnn/test_mkldnn_op_nhwc.cc [hunk 1]
```diff
@@ -28,7 +28,7 @@
 #include "paddle/phi/core/kernel_registry.h"
 
 USE_OP_ITSELF(pool2d);
-USE_OP_DEVICE_KERNEL(pool2d, MKLDNN);
+PD_DECLARE_KERNEL(pool2d, OneDNN, ALL_LAYOUT);
```
Comment (Silv3S) [inline paddle/fluid/operators/mkldnn/test_mkldnn_op_nhwc.cc]:
```suggestion
PD_DECLARE_KERNEL(pool2d, OneDNN, ONEDNN);
```

**Review by Silv3S — DISMISSED** [review]
LGTM

File: paddle/phi/kernels/onednn/pool_grad_kernel.cc [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  }
+  astream.wait();
+
+  dx->set_mem_desc(diff_src_memory->get_desc());
+}
+}  // namespace phi
+
+PD_REGISTER_KERNEL(pool2d_grad,
+                   OneDNN,
+                   ALL_LAYOUT,
```
Comment (Silv3S) [inline paddle/phi/kernels/onednn/pool_grad_kernel.cc]: Sorry, didn't catch this earlier
```suggestion
                   ONEDNN,
```

File: paddle/phi/kernels/onednn/pool_kernel.cc [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  }
+  astream.wait();
+
+  out->set_mem_desc(dst_memory->get_desc());
+}
+}  // namespace phi
+
+PD_REGISTER_KERNEL(pool2d,
+                   OneDNN,
+                   ALL_LAYOUT,
```
Comment (Silv3S) [inline paddle/phi/kernels/onednn/pool_kernel.cc]:
```suggestion
                   ONEDNN,
```

**Review by YuanRisheng — APPROVED** [review]

**Review by sfraczek — COMMENTED** [review]
LGTM

**Pull request merged**